### ISSUE TITLE
The map in dark mode is unusable

### ISSUE THREAD

**Issue by mirg** [home-assistant/frontend#9493]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The map in dark mode is unusable, no contrast at all.

### Describe the behavior you expected

The map in dark mode should be visible as in normal (non-dark) mode.

### Steps to reproduce the issue

1. set dark mode
2. open map

### What version of Home Assistant Core has the issue?

core-2021.6.5

**Comment by spacegaier** [comment]
The map is not something we construct ourselves, see here: https://github.com/home-assistant/frontend/issues/6622#issuecomment-679886153

Buildings in the dark mode are indeed difficult to see.
Not sure about the comment from skandalfo.

**Comment by bramkragten** [comment]
If someone finds some nice dark mode tiles for Leaflet that we are allowed to use, we are very much open to that.

**Comment by skandalfo** [comment]
https://stackoverflow.com/a/37618411 points to [leaflet-tilefilter](https://github.com/humangeo/leaflet-tilefilter) which is MIT-licensed.

E.g.: http://humangeo.github.io/leaflet-tilefilter/demo.html and choose OSM Default maps, then combine "Invert" in the canvas filter with "HueRotate180" in the CSS filter for a quick example of how to invert brightness without changing the colours?

**Comment by bramkragten** [comment]
Yes, I know it can be done with CSS filters, but that does require extra CPU from the users' client, which I would like to avoid if possible.

**Comment by skandalfo** [comment]
ACK. Just saying I'll be happy to not have to squint hard at my phone to try to see the streets and labels in daylight, even if that means paying a bit more in battery usage for a pixel-local color correction pass on each draw. Even just having the option to switch to OSM standard without regard to the current theme would be a huge usability improvement for me.

If that makes any difference (I have no idea), the leaflet-tilefilter thing above also offers the option to perform the filtering at the "canvas" level instead of the CSS level.

**Comment by bramkragten** [comment]
An option for a tile provider might be an option, another option could be a theme var where you can input a filter, but that would probably also need the tile provider option, as I think a filter doesn't work well on the dark tiles?
